_custom_build/backend.py

```diff
@@ -0,0 +1,56 @@
+import sys
+
+from setuptools.build_meta import *  # noqa: F401, F403
+from setuptools.build_meta import build_wheel
+
+backend_class = build_wheel.__self__.__class__
+
+
+class _CustomBuildMetaBackend(backend_class):
+    def run_setup(self, setup_script="setup.py"):
+        if self.config_settings:
+
+            def config_has(key, value):
+                settings = self.config_settings.get(key)
+                if settings:
+                    if not isinstance(settings, list):
+                        settings = [settings]
+                    return value in settings
+
+            flags = []
+            for dependency in (
+                "zlib",
+                "jpeg",
+                "tiff",
+                "freetype",
+                "raqm",
+                "lcms",
+                "webp",
+                "webpmux",
+                "jpeg2000",
+                "imagequant",
+                "xcb",
+            ):
+                if config_has(dependency, "enable"):
+                    flags.append("--enable-" + dependency)
+                elif config_has(dependency, "disable"):
+                    flags.append("--disable-" + dependency)
+            for dependency in ("raqm", "fribidi"):
+                if config_has(dependency, "vendor"):
+                    flags.append("--vendor-" + dependency)
+            if self.config_settings.get("platform-guessing") == "disable":
+                flags.append("--disable-platform-guessing")
+            if self.config_settings.get("debug") == "true":
+                flags.append("--debug")
+            if flags:
+                sys.argv = sys.argv[:1] + ["build_ext"] + flags + sys.argv[1:]
+        return super().run_setup(setup_script)
+
+    def build_wheel(
```

Comment:
@hugovk @radarhere this omits `build_editable()` which is almost the same as `build_wheel()`. You'll likely want to wrap it as well.

Reply:
I've attempted to add `build_editable` in https://github.com/python-pillow/Pillow/commit/290a2d15b6066e66b797b985bdedf6986c271f41, but for some reason it is not working properly on Windows: https://github.com/nulano/Pillow/actions/runs/7308594440/job/19915362677#step:25:728

Reply:
It looks like the build script calls something different from what's called on the main branch: I see "Running command Getting requirements to build editable" vs. "Running command Getting requirements to build wheel". The former is from your log, the latter is from the main branch.

Reply:
Yes, I added https://github.com/python-pillow/Pillow/commit/2446fd8d7ca7d495ed19addefd28e2925c74e629 to test editable installing on GHA. For some reason, it seems the include paths are not passed in properly when building editable on Windows (specifically the OpenJpeg path is found and added for part of the build, but not all of it).

Reply:
Oh, interesting, I didn't see where it's defined. It's often hard to distinguish outputs of different commands if they are squashed in the same CI step, so I tend to recommend putting each command in a separate step or using the corresponding shell's capabilities to improve such logging. I also recently saw a case where Powershell would proceed to execute commands after the previous ones fail.

Anyway, this may indicate dependency on some weird pre-existing state.

One way of exploring GHA jobs is https://github.com/marketplace/actions/debugging-with-tmate, by the way.

Reply:
Sorry, I should have mentioned that it also fails for me locally.

It's the first command in that step that is failing, the second one just runs a simple test.
The path is added in `setup.py` in the `build_extensions` method here: https://github.com/python-pillow/Pillow/blob/41e45b5f95c5184b8b4c4712e5614bf468331e05/setup.py#L687-L691
From the GHA log, it looks almost as if it first does a regular build, then a second build again skipping that method, but I'm not sure why or where to look.

Yes, I have used tmate before.

Reply:
I've created #7658 for this.